File: android/app/src/main/kotlin/com/bockmedia/console/domain/model/SessionDataStore.kt
package com.bockmedia.console.domain.model

import com.bockmedia.console.data.api.dto.AnalyticsResponse
import com.bockmedia.console.data.api.dto.DashboardQuickResponse
import com.bockmedia.console.data.api.dto.PlaylistsResponse
import com.bockmedia.console.data.api.dto.StreamHistoryResponse

/** Cross-tab session cache — Home, Library, and Search share one playlist/history snapshot. */
object SessionDataStore {
    private const val TTL_MS = 10 * 60 * 1000L

    private var playlistsKey: String? = null
    private var playlistsAtMs: Long = 0
    private var playlists: PlaylistsResponse? = null

    private var historyAtMs: Long = 0
    private var history: StreamHistoryResponse? = null

    private var dashboardAtMs: Long = 0
    private var dashboard: DashboardQuickResponse? = null

    private var analyticsAtMs: Long = 0
    private var analytics: AnalyticsResponse? = null

    fun peekPlaylists(memberKey: String): PlaylistsResponse? {
        if (playlistsKey != memberKey || System.currentTimeMillis() - playlistsAtMs > TTL_MS) return null
        return playlists
    }

    fun putPlaylists(memberKey: String, response: PlaylistsResponse) {
        playlistsKey = memberKey
        playlistsAtMs = System.currentTimeMillis()
        playlists = response
    }

    fun invalidatePlaylists() {
        playlistsKey = null
        playlists = null
        playlistsAtMs = 0
    }

    fun peekHistory(): StreamHistoryResponse? =
        if (System.currentTimeMillis() - historyAtMs <= TTL_MS) history else null

    fun putHistory(response: StreamHistoryResponse) {
        historyAtMs = System.currentTimeMillis()
        history = response
    }

    fun peekDashboard(): DashboardQuickResponse? =
        if (System.currentTimeMillis() - dashboardAtMs <= TTL_MS) dashboard else null

    fun putDashboard(response: DashboardQuickResponse) {
        dashboardAtMs = System.currentTimeMillis()
        dashboard = response
    }

    fun peekAnalytics(): AnalyticsResponse? =
        if (System.currentTimeMillis() - analyticsAtMs <= TTL_MS) analytics else null

    fun putAnalytics(response: AnalyticsResponse) {
        analyticsAtMs = System.currentTimeMillis()
        analytics = response
    }

    fun invalidateAll() {
        invalidatePlaylists()
        history = null
        historyAtMs = 0
        dashboard = null
        dashboardAtMs = 0
        analytics = null
        analyticsAtMs = 0
    }
}

File: ios/BockMedia/Domain/SessionDataStore.swift
import Foundation

/// Cross-tab session cache — Home, Library, and Search share one playlist/history snapshot.
enum SessionDataStore {
    private static let ttl: TimeInterval = 10 * 60

    private static var playlistsKey: String?
    private static var playlistsAt: Date?
    private static var playlists: PlaylistsResponse?

    private static var historyAt: Date?
    private static var history: StreamHistoryResponse?

    private static var dashboardAt: Date?
    private static var dashboard: DashboardQuickResponse?

    private static var analyticsAt: Date?
    private static var analytics: AnalyticsResponse?

    static func peekPlaylists(memberKey: String) -> PlaylistsResponse? {
        guard playlistsKey == memberKey,
              let at = playlistsAt,
              Date().timeIntervalSince(at) < ttl else { return nil }
        return playlists
    }

    static func putPlaylists(memberKey: String, response: PlaylistsResponse) {
        playlistsKey = memberKey
        playlistsAt = Date()
        playlists = response
    }

    static func invalidatePlaylists() {
        playlistsKey = nil
        playlistsAt = nil
        playlists = nil
    }

    static func peekHistory() -> StreamHistoryResponse? {
        guard let at = historyAt, Date().timeIntervalSince(at) < ttl else { return nil }
        return history
    }

    static func putHistory(_ response: StreamHistoryResponse) {
        historyAt = Date()
        history = response
    }

    static func peekDashboard() -> DashboardQuickResponse? {
        guard let at = dashboardAt, Date().timeIntervalSince(at) < ttl else { return nil }
        return dashboard
    }

    static func putDashboard(_ response: DashboardQuickResponse) {
        dashboardAt = Date()
        dashboard = response
    }

    static func peekAnalytics() -> AnalyticsResponse? {
        guard let at = analyticsAt, Date().timeIntervalSince(at) < ttl else { return nil }
        return analytics
    }

    static func putAnalytics(_ response: AnalyticsResponse) {
        analyticsAt = Date()
        analytics = response
    }

    static func invalidateAll() {
        invalidatePlaylists()
        historyAt = nil
        history = nil
        dashboardAt = nil
        dashboard = nil
        analyticsAt = nil
        analytics = nil
    }
}
